### ISSUE TITLE
Frequent CMake version requirement changes

### ISSUE THREAD

**Issue by kostrzewa** [lattice/quda#846]
I wanted to inquire about the frequent CMake version requirement changes and whether these are absolutely necessary. Most sysadmins, it seems to me, are very conservative and getting them to install packages more recent than ~ 2 years is difficult. This is doubly true of many supercomputer installations, where software versions sometimes remain constant (or are only mildly modified) over the life of the machine... At more flexible installations, hierarchical module systems sometimes mean that a particular combination of software versions cannot be activated due to unfulfillable interdependencies. On the other hand, I do know of installations where CMake versions are kept reasonably up to date, so it really depends where one has to run.

I'm aware that one can install CMake binaries locally relatively easily, but that becomes quite a hassle when dealing with ever increasing numbers of machines simultaneously, especially if problems arise.  As a developer, I fully agree with following CMake development versions reasonably closely, but some people have a hard time as it is to get everything set up, such that adding steps to the installation procedure results in support mails for those of us dealing with "end users"....

Looking at the commits that introduced the 3.11 requirement, I don't really see anything that mandated the version requirement change (please correct me if I'm wrong as I haven't checked any of the new features). It seems that I'm able to compile the library just fine with CMake 3.9.2 after changing the requirement in CMakeLists.txt.

Just a few thoughts...

**Comment by mathiaswagner** [comment]
There are technical reasons. And since you can download cmake binaries for Linux from cmake.org that only require you to unpack a tar in your home it should not prevent anyone from compiling QUDA. 
Most systems these days have reasonably recent cmake modules. And 3.11 is over a year old. 

Tl;dr we are aware and don’t change requirements for no good reason. Since cmake is now the only option we will add information on how to get a cmake and how to use it with QUDA to the wiki. 

Does that work for you?

**Comment by mathiaswagner** [comment]
We also have things in mind that we want to add in the future that will require 3.11 and so to not bump it up frequently we went with 3.11.

**Comment by kostrzewa** [comment]
Understood. Wasn't meant as criticism, just something that I and others keep bumping into on various machines.

> We also have things in mind that we want to add in the future that will require 3.11 and so to not bump it up frequently we went with 3.11.

Makes sense!

**Comment by mathiaswagner** [comment]
I know. Brevity is just me typing on my phone. We try to stay compatible but there will always be some pain points when moving forward.

**Comment by mathiaswagner** [comment]
Follow up issue #847
